File: frontend/src/types/purchase.ts
export type PurchaseType = {
  id: number
  datetime: string
  datetimeDisplay: boolean
  total: number
  title: string
  image: string
  category: string
  category_name: string
  price: number
}

export type PurchasesType = {
  quantity: number
  purchases: Array<PurchaseType>
}

export type PurchaseHistoryItemType = {
  image: string
  category_name: string
  title: string
  price: number
}

export type PurchaseHistoryType = Array<{
  datetime: string
  total: number
  items: Array<PurchaseHistoryItemType>
}>

File: backend/app/types/purchase.go
package types

type APIPurchase struct {
	ID           uint64 `json:"id"`
	Datetime     string `json:"datetime"`
	Total        uint64 `json:"total"`
	Title        string `json:"title"`
	Image        string `json:"image"`
	Category     string `json:"category"`
	CategoryName string `json:"category_name"`
	Price        uint64 `json:"price"`
}

type APIPurchases struct {
	Quantity  int           `json:"quantity"`
	Purchases []APIPurchase `json:"purchases"`
}
